pkg/controllers/dynakube/extension/secret.go

```diff
@@ -0,0 +1,83 @@
+package extension
+
+import (
+	"context"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/dttoken"
+	k8ssecret "github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/secret"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	eecTokenKey         = "eec-token"
+	eecTokenValuePrefix = "EEC dt0x01"
+	secretSuffix        = "-extensions-token"
+)
+
+func reconcileSecret(ctx context.Context, dk *dynakube.DynaKube, kubeClient client.Client, apiReader client.Reader) error {
+	log.Info("reconciling secret")
+
+	query := k8ssecret.NewQuery(ctx, kubeClient, apiReader, log)
+
+	if !dk.PrometheusEnabled() {
+		err := query.Delete(getSecretName(dk.Name), dk.Namespace)
+		if err != nil {
+			return err
+		}
+
+		removeSecretCreated(dk.Conditions())
+
+		return nil
+	}
+
+	_, err := query.Get(client.ObjectKey{Name: getSecretName(dk.Name), Namespace: dk.Namespace})
+	if err != nil && !errors.IsNotFound(err) {
+		setSecretCreatedFalse(dk.Conditions(), err)
+
+		return err
+	}
+
+	if errors.IsNotFound(err) {
```

Comment:
why are we not checking if the existing secret contains what we expect it to?

Reply:
AFAIK we generate its content (the eec-token) and we can't validate it against the API for now, but I will talk with the team about it.